Regression: `line_length` option `ignores_function_declarations` no longer applies to initializers

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Previously, with SwiftLint **0.59.1**, this code does not trigger a `line_length` warning when `ignores_function_declarations: true`:
```swift
public class FileDataStore {
    public init(persistenceService: PersistenceService, syncService: SyncService, notificationService: NotificationService) throws {
        // ...
    }

    public func doSomething(persistenceService: PersistenceService, syncService: SyncService, notificationService: NotificationService) throws {
        // ...
    }
}
```

```
$ swiftlint Modules/Sources/Stores/DataStore/FileDataStore/FileDataStore.swift
Linting Swift files at paths Modules/Sources/Stores/DataStore/FileDataStore/FileDataStore.swift
Linting 'FileDataStore.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

After upgrading to SwiftLint **0.61.0**, the `line_length` rule (still with `ignores_function_declarations: true`) now warns about a violation on the `public init` line, but not on the `public func` line.

```
$ swiftlint Modules/Sources/Stores/DataStore/FileDataStore/FileDataStore.swift
Linting Swift files at paths Modules/Sources/Stores/DataStore/FileDataStore/FileDataStore.swift
Linting 'FileDataStore.swift' (1/1)
/[…]/Modules/Sources/Stores/DataStore/FileDataStore/FileDataStore.swift:2:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 132 characters (line_length)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure): _**0.61.0**_
* Xcode version (run `xcodebuild -version` to be sure): _**Xcode 16.4 (Build version 16F6)**_
* Installation method used (Homebrew, CocoaPods, building from source, etc): _**Homebrew**_
* Configuration file:

```yml
only_rules:
  - line_length

line_length:
  ignores_function_declarations: true
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? _**No**_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,7 +16,10 @@
 
 ### Bug Fixes
 
-* None.
+* Ignore function, initializer and subscript declarations alike when the
+  `ignores_function_declarations` option is enabled in the `line_length` rule.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#6241](https://github.com/realm/SwiftLint/issues/6241)
 
 ## 0.61.0: Even Fresher Breeze
 

--- a/Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift
@@ -147,6 +147,18 @@ private final class FunctionLineVisitor: SyntaxVisitor {
     }
 
     override func visitPost(_ node: FunctionDeclSyntax) {
+        collectLines(from: node)
+    }
+
+    override func visitPost(_ node: InitializerDeclSyntax) {
+        collectLines(from: node)
+    }
+
+    override func visitPost(_ node: SubscriptDeclSyntax) {
+        collectLines(from: node)
+    }
+
+    private func collectLines(from node: any SyntaxProtocol) {
         let startLocation = locationConverter.location(for: node.positionAfterSkippingLeadingTrivia)
         let endLocation = locationConverter.location(for: node.endPositionBeforeTrailingTrivia)
         for line in startLocation.line...endLocation.line {

--- a/Tests/BuiltInRulesTests/LineLengthRuleTests.swift
+++ b/Tests/BuiltInRulesTests/LineLengthRuleTests.swift
@@ -7,12 +7,35 @@ final class LineLengthRuleTests: SwiftLintTestCase {
             "c: String, d: String, e: String, f: String, g: String, h: String, i: String, " +
             "j: String, k: String, l: String, m: String, n: String, o: String, p: String, " +
             "q: String, r: String, s: String, t: String, u: String, v: String, w: String, " +
-            "x: String, y: String, z: String) {\n"),
+            "x: String, y: String, z: String) {}\n"),
         Example("func superDuperLongFunctionDeclaration(a: String, b: String, " +
             "c: String, d: String, e: String, f: String, g: String, h: String, i: String, " +
             "j: String, k: String, l: String, m: String, n: String, o: String, p: String, " +
             "q: String, r: String, s: String, t: String, u: String, v: String, w: String, " +
-            "x: String, y: String, z: String) {\n"),
+            "x: String, y: String, z: String) {}\n"),
+        Example("""
+            struct S {
+                public init(a: String, b: String, c: String, d: String, e: String, f: String, \
+                            g: String, h: String, i: String, j: String, k: String, l: String, \
+                            m: String, n: String, o: String, p: String, q: String, r: String, \
+                            s: String, t: String, u: String, v: String, w: String, x: String, \
+                            y: String, z: String) throws {
+                    // ...
+                }
+            }
+            """),
+        Example("""
+            struct S {
+                subscript(a: String, b: String, c: String, d: String, e: String, f: String, \
+                          g: String, h: String, i: String, j: String, k: String, l: String, \
+                          m: String, n: String, o: String, p: String, q: String, r: String, \
+                          s: String, t: String, u: String, v: String, w: String, x: String, \
+                          y: String, z: String) -> Int {
+                    // ...
+                    return 0
+                }
+            }
+            """),
     ]
 
     private let longComment = Example(String(repeating: "/", count: 121) + "\n")
@@ -50,8 +73,12 @@ final class LineLengthRuleTests: SwiftLintTestCase {
         let nonTriggeringExamples = baseDescription.nonTriggeringExamples + longFunctionDeclarations
         let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
 
-        verifyRule(description, ruleConfiguration: ["ignores_function_declarations": true],
-                   commentDoesntViolate: false, stringDoesntViolate: false)
+        verifyRule(
+            description,
+            ruleConfiguration: ["ignores_function_declarations": true],
+            commentDoesntViolate: false,
+            stringDoesntViolate: false
+        )
     }
 
     func testLineLengthWithIgnoreCommentsEnabled() {

